---
title: "Drilling COSC-2 microbiology"
subtitle: "Characterization of microbial communities"
author: "George"
date: "2023-02-15"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```


```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
# Ranked in order of abundance
cols.phylum <- c(
  "Firmicutes" = "#A2A475",
  "Desulfobacterota" = "#02401B",
  "Actinobacteriota" = "#D69C4E",
  "Proteobacteria" = "#972D15",
  "Other" = "#899DA4"
  )

# Ranked in order of abundance
cols.taxon <- c(
  "Desulforudis" = "#A2A475",
  "Desulfitibacter" = "#FAEFD1",
  "Smithellaceae" = "#02401B",
  "Coriobacteriia" = "#D69C4E",
  "Pelotomaculaceae" = "#00A08A",
  "Dethiosulfatibacteraceae" = "#F2AD00",
  "Erysipelotrichaceae" = "#046C9A",
  "Other" = "#899DA4"
  )

cols.class <- c(
  "Desulfotomaculia" = "#A2A475",
  "Moorellia" = "#FAEFD1",
  "Clostridia" = "#00A08A",
  "Bacteroidia" = "#02401B",
  "Bacilli" = "#D69C4E",
  "Gammaproteobacteria" = "#972D15",
  "Alphaproteobacteria" = "#046C9A",
  "Other" = "#899DA4"
  ) %>% rev()

cols.euk <- c(
  "Metazoa" = "#A2A475",
  "Ochrophyta" = "#FAEFD1",
  "Chlorophyta" = "#00A08A",
  "Dinoflagellata" = "#02401B",
  "Cercozoa" = "#972D15",
  "Fungi" = "#046C9A",
  "Other" = "#899DA4"
) %>% rev()
```


# Read input files


```{r read input files 16S}
# GTDB release 207 was used as reference database

seqtab <- read_tsv("ASV_table.tsv", col_types = cols(.default = col_character()))
gtdb <- read_tsv("ASV_tax_species.tsv", col_types = cols(.default = col_character()))

# The three samples for COSC-2 are 61, 62, and 63, 64 is negative extraction control
# Each sample contained two filters that were pooled prior to DNA extraction
# The extraction control contains 43 unique ASVs
i <- c("P28309_1061","P28309_1062","P28309_1063","P28309_1064")

# Add relative abundances
seqtab <- seqtab %>%
  pivot_longer(-1, names_to = "sample", values_to = "count") %>%
  mutate(sample = gsub("_S[0-9].", "", sample)) %>%
  filter(sample %in% i) %>% 
  mutate(count = as.integer(count)) %>% 
  filter(count > 0) %>%
  group_by(sample) %>% 
  mutate(relab = count / sum(count)) %>% 
  ungroup()

# Extract the ASVs from the extraction control
c <- seqtab %>% filter(sample == "P28309_1064") %>% pull(seqid)

gtdb <- gtdb %>% mutate(taxon = coalesce(genus,family,order,class,phylum,domain)) %>% select(-kingdom)
```




```{r read meta-g}
# CoverM was used for mapping the trimmed reads to the genomes
# This file only contains the bins for this project (n = 27)
coverm <- read_tsv("coverm.tsv", col_types = cols(.default = col_character(), tpm = col_double()))

# Before running CheckM with the CPR market set, this objected contained 567 genomes
# After running CheckM with CPR markers 570 genomes
checkm <- read_tsv("bin-stats-checkm.tsv", show_col_types = F)
checkm <- checkm %>% filter(genome %in% coverm$genome)

# Taxonomy was assigned to bins with completeness > 50% and contamination > 5%
# Average completeness and contamination of Patescibacteria before CPR marker set is 65.8% (sd 7.09, n = 12) and 1.17% (sd 1.38)
# Using CPR market set these numbers are 91.8% (sd 7.95, n = 15) and 0.930 (sd 1.71)


bintax <- read_tsv("gtdbtk.summary.tsv", show_col_types = F) %>%
  select(genome = user_genome, classification) %>%
  mutate(classification = gsub("[a-z]__", "", classification)) %>%
  separate(classification, 
           into = c("domain","phylum","class","order","family","genus","species"), 
           sep = ";", fill = "right") %>%
  mutate(across(everything(), na_if, "")) %>%
  mutate(across(-1, \(x) gsub(x, pattern = "_[A-Z]", replacement = ""))) %>%
  filter(genome %in% coverm$genome) %>% distinct() %>%
  # Remove placeholder names
  mutate(across(2:8, \(x) ifelse(grepl("[0-9]", x), NA, x))) %>%
  mutate(taxon = coalesce(genus, family, order, class, phylum))
```


```{r read files emapper, prokka, and keggmapper}
emapper <- data.frame()

for (file in list.files("emapper/", full.names = TRUE)) { # List all files in the emapper directory
  name <- substr(file, 10, nchar(file)) %>% gsub("emapper.emapper.annotations", "", .) # Extract the bin ref from the filename
  i <- read_tsv(file, show_col_types = F, comment = "##") # Read the file
  i <- i %>% rename(query = 1) %>% mutate(genome = name) %>% distinct() %>% select(genome, everything()) %>% mutate(KEGG_ko = gsub("ko:", "", KEGG_ko))
  emapper <- emapper %>% rbind(i)
}

prokka <- data.frame()
for (file in list.files("prokka/", full.names = TRUE)) {
  name <- substr(file, 9, nchar(file)) %>% gsub(".tsv", "", .)
  i <- read_tsv(file, show_col_types = F, comment = "##")
  i <- i %>% mutate(genome = name) %>% distinct() %>% 
    select(genome, everything()) %>% 
    select(-locus_tag) %>% 
    rename(ec = EC_number)
  prokka <- prokka %>% rbind(i)
}

kegg <- read_tsv("keggmapper.tsv", col_types = cols(.default = col_character()))
```

# Extraction control (supplemental)

```{r community with control}
seqtab %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(class, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(class)) %>%
  top_n(7, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(class, topphylum = class), by = "class") %>%
  #mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

p1 <- seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, names(cols.class))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Class") +
  geom_col() +
  scale_fill_manual(values = cols.class) +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  4.45, y = 101, yend = 101) +
  annotate('text', x =  2, y = 104, size = 7 / .pt, label = "COSC filters", hjust = 0.5) +
  annotate('text', x =  4, y = 104, size = 7 / .pt, label = "Extraction c.", hjust = 0.5) +
  theme_tidy(legend = "bottom") + guides(fill = guide_legend(reverse = T, ncol = 2)) +
  theme(
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    legend.key.height = unit(4, "mm")
  )
```

```{r nmds with controls}
meta <- list(sample = c("P28309_1061","P28309_1062","P28309_1063","P28309_1064"),
     type = c("Sample #1","Sample #2","Sample #3","Neg. ex. control"))

set.seed(999)
nmds <- seqtab %>% # Full size fraction
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>% vegan::metaMDS()

vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") -> nmds.scores

p2 <- nmds.scores %>%
  inner_join(meta, by = "sample", copy = T) %>%
  ggplot(aes(NMDS1, NMDS2)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  # Points for samples, coloured by nature
  geom_point(size = 3) +
  theme_tidy() +
  ggrepel::geom_label_repel(aes(label = type), color = "black",
                  size = 7/.pt) +
  annotate('text', x = Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = 1.2, vjust = -1,
           )
```


```{r combine plots with control, fig.width=18/2.53}
library(patchwork)
p1 + p2 + p3 + plot_layout(nrow = 2)
```

```{r export control plot}
ggsave("../figures/control.pdf", width = 16, height = 16, units = "cm")
```

# Field site

```{r plot scandinavia}
# Äspö Lat N 57.4334, Lon E 16.6603) 
baltic <- list.files("../../Cultures/didactic-spork/data/shapefiles/", pattern = ".shp", full.names = TRUE) %>% lapply(sf::st_read)
# Use EPSG 3152, unit (m)
baltic[[3]] <- baltic[[3]] %>% sf::st_transform(3152)
baltic[[6]] <- baltic[[6]] %>% sf::st_transform(3152)
baltic[[7]] <- baltic[[7]] %>% sf::st_transform(3152)

ggplot() + # x (longitude), y (latitude)
  geom_sf(data = baltic[[3]], colour = "black", fill = "white", lwd = 0.08) + # Finland
  geom_sf(data = baltic[[6]], colour = "black", fill = "white", lwd = 0.08) + # Norway
  geom_sf(data = baltic[[7]], colour = "black", fill = "white", lwd = 0.08) + # Sweden
  ggspatial::annotation_scale(location = "br") +
  annotate("rect", xmin = -126780 - 20000, xmax = -126780 + 20000, ymin = 530561 - 20000, ymax = 530561 + 20000, fill = "black" 
           ) +
  annotate("text", x = -126780 + 110000, y = 530561, label = "COSC-2", colour = "black", size = 7/.pt) +
  lims(x = c(-605927, 377430), y = c(-300000, 1172471)) +
  labs(x = "", y = "") + 
  theme(
    panel.grid = element_line(colour = "#D9D9D9", linewidth = 0.4),
    panel.background = element_blank(),
    axis.text = element_text(size = 7, color = "black"),
    axis.ticks = element_blank()
    )
```

```{r export map}
ggsave("../figures/coscmap.png", width = 12, height = 10, units = "cm")
```


# Bar plot


```{r plot community phylum}
i <- c("VK-3516-f1","VK-3516-f3","VK-3516-f5","P28309_1061","P28309_1062","P28309_1063")

p1 <- coverm %>%
  # Prepare the data to be merged with the amplicon data
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(phylum = ifelse(!is.na(phylum) & phylum %in% names(cols.phylum), phylum, "Other")) %>%
  # Sum abundance within phylum
  group_by(sample, phylum) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(phylum = factor(phylum, levels = names(cols.phylum))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, 
             fill = fct_rev(phylum))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  labs(x = "", y = "Relative abundance (%)") +
  scale_fill_manual(values = rev(cols.phylum), name = "") +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  0.55, y = Inf, label = "bold(a)", 
           size = 8/.pt, hjust = 0, vjust = 1, parse = TRUE) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt)
```


```{r plot community highest resolution}
p2 <- coverm %>%
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(taxon = case_when(
    genus %in% c("Desulforudis","Desulfitibacter") ~ genus,
    family %in% c("Smithellaceae","Pelotomaculaceae","Dethiosulfatibacteraceae",
                  "Erysipelotrichaceae") ~ family,
    class == "Coriobacteriia" ~ class,
    .default = "Other")
    ) %>%
  # Sum abundance within phylum
  group_by(sample, taxon) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(taxon = factor(taxon, levels = names(cols.taxon))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, fill = fct_rev(taxon))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank()
        ) +
  labs(x = "", y = "") +
  scale_fill_manual(values = cols.taxon, name = "", 
                    labels = c("Other","Erysipelotrichaceae(f)",
                               "Dethiosulfatibacteraceae (f)","Pelotomaculaceae (f)",
                               "Coriobacteriia (c)","Smithellaceae (f)", 
                               "Desulfitibacter (g)","D. audaxviator (s)"
                               )) +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt) +
  annotate("text", x =  0.55, y = Inf, label = "bold(b)", 
           size = 8/.pt, hjust = 0, vjust = 1, parse = TRUE)
```


```{r combine both levels, fig.width=18/2.53}
library(patchwork)
p1 + p2 + p3 & guides(fill = guide_legend(ncol = 2, reverse = TRUE)) & 
  theme(
  legend.key.height = unit(4, "mm"),
  legend.spacing = margin(),
  legend.margin = margin(-12,0,-12,0),
  legend.box.margin = margin(),
  plot.margin = margin(),
  aspect.ratio = 1.5,
  legend.position = "bottom",
  panel.border = element_blank(),
  axis.line = element_line(linewidth = 0.4, colour = "black"),
  axis.title.y = element_text(margin = margin(0,-2,0,0))
)
```


```{r export community plot phylum}
ggsave("../figures/barplot_euk.pdf", width = 18, height = 12, units = "cm")
```


# Visualise stats of genomes (Fig. 3)


```{r}
i <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")
# Desulforudis audaxviator GC 0.6085068, genome size 2616697 bp
library(ggtext)

p1 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, GC, genome.size) %>%
  distinct() %>%
  mutate(genome.size = genome.size/1e6) %>%
  ggplot(aes(x = genome.size, 
             y = GC, 
             color = fct_relevel(phylum,i)
             )) +
  geom_smooth(method = "lm", formula = y ~ x, colour = "black", fill = "#d9d9d9", se = TRUE, linetype = 0) +
  geom_point(size = 2.5, shape = 1, stroke = 1) + 
  annotate("text", x = Inf, y = -Inf, 
           label = "paste(R ^ 2, \" = 0.06\")", 
           size = 7/.pt, vjust = -0.5, hjust = 1, parse = TRUE) +
  annotate("richtext", x = 2.8, y = 0.609, 
           label = "D. audaxviator: 2.6 Mbp", 
           size = 7/.pt, hjust = 0, label.color = "#A2A475") +
  labs(x = "Estimated genome size (Mbp)", y = "GC content (%)", color = "") +
  scale_x_continuous(breaks = c(1,2,3,4,5,6)) +
  scale_color_manual(values = cols.phylum, guide = "none") +
  theme_tidy() +
  theme(
    text = element_text(colour = "black"),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, "black")
  )
```



```{r plot bin stats, fig.width = 14/2.54}
names <- c(
  "completeness" = "Completeness (%)",
  "predicted.genes" = "No. predicted genes"
)

p2 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, completeness) %>%
  distinct() %>%
  ggplot(aes(x = fct_relevel(phylum,i), 
             y = completeness, 
             group = phylum, 
             colour = phylum, 
             fill = fct_relevel(phylum,i)
             )) +
  geom_boxplot(alpha = 0.5, key_glyph = "rect") +
  labs(x = "", y = "Completeness (%)", fill = "") +
  scale_colour_manual(values = cols.phylum, guide = "none") +
  scale_fill_manual(values = cols.phylum) +
  theme_tidy(legend = "bottom", ratio = 1.2) +
  theme(
    axis.line = element_line(linewidth = 0.4, "black"),
    panel.border = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    legend.key.height = unit(4, "mm"),
    axis.title.y = element_text(margin = margin()),
    plot.margin = margin()
  )
```


```{r ordination on function metagenomes}
nmds <- coverm %>%
  inner_join(emapper, by = "genome") %>% 
  filter(KEGG_ko != "-") %>% # Filter out NA
  group_by(genome, KEGG_ko) %>% summarise(tpm = sum(tpm), .groups = "drop") %>%
  pivot_wider(names_from = "KEGG_ko", values_from = "tpm", values_fill = 0) %>%
  column_to_rownames("genome") %>%
  vegan::metaMDS()

nmds.scores <- vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample")
```


```{r plot ordination KOs}
library(ggtext)
i <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")

p3 <- nmds.scores %>%
  rename(genome = sample) %>% inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  # Plot
  ggplot(aes(x = NMDS1, 
             y = NMDS2, 
             colour = fct_relevel(phylum,i))) +
  geom_vline(xintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  geom_hline(yintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  # Points for samples, coloured by nature
  geom_point(size = 2.5, shape = 1, stroke = 1) +
  theme_tidy() +
  scale_color_manual(values = cols.phylum, name = "", guide = "none") +
  annotate('text', x = Inf, y = -Inf, size = 7/.pt,
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = 1.05, vjust = -0.5) +
  annotate("richtext", x = -2.9, y = 3.69, 
           label = "Methanobacteriota", 
           size = 7/.pt, hjust = 0, label.color = "#899DA4", 
           label.padding = unit(c(0.1,0.1,0.1,0.1), "lines")) +
  annotate("richtext", x = -4.5, y = -2.57, 
           label = "Bradyrhizobium", 
           size = 7/.pt, hjust = 0, label.color = "#972D15", 
           label.padding = unit(c(0.1,0.1,0.1,0.1), "lines")) +
  theme(
    axis.title.y = element_text(margin = margin()),
    panel.background = element_blank(),
    panel.border = element_rect(linewidth = 0.4, colour = "black")
    )
```


```{r plot stats on bins, fig.width = 18/2.54}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") + plot_annotation(tag_levels = "a") & 
  theme(
    plot.tag.position = c(0.15, 1.05),
    plot.tag = element_text(size = 8, colour = "black", face = "bold"),
    legend.position = "bottom",
    plot.margin = margin(),
    plot.background = element_blank())
```


```{r export bin stats}
ggsave("../figures/binstats.pdf", height = 10, width = 18, units = "cm")
```


# Heatmap

```{r order taxa}
# Taxa are ordered (descending) based on phylum and tpm
# Only taxa are included with tpm > 10,000 (relative abundance > 1 %)
# 14 / 27 genomes are included
names <- coverm %>% 
  group_by(genome) %>% summarise(tpm = sum(tpm)) %>% 
  filter(tpm > 10000) %>% inner_join(bintax, by = "genome") %>%
  arrange(desc(tpm)) %>% pull(genome)
```


```{r multiple gene copies}
emapper %>%
  rename(ec = EC, gene = Preferred_name) %>%
  mutate(category = case_when(
    # S metabolism
    ec == "2.7.7.4" ~ "03: Sulfate adenylyltransferase (sat)", # sat
    grepl("apr[A,B]", gene) ~ "04: Adenylylsulfate reductase (aprAB)", # aprAB
    grepl("asr[A-C]", gene) ~  "05: Anaerobic sulfite reductase (asrABC)", # asrABC
    grepl("sox[X,Y,Z,B]", gene) ~ "07: Thiosulfate ox. (soxABXYZ)", # soxA: soxBCD not present
    gene == "soeB" ~ "08: Sulfite dehydrogenase (soeB)", # soeB: soeAC not present
    gene == "sorB" ~ "08: Sulfite dehydrogenase (sorB)",
    # N metabolism
    ec == "1.18.6.1" ~ "09: Nitrogen fixation (nifKDH)", # nifKDH
    grepl("nrf[A,H]", gene) ~ "13: Dissimilatory nitrate red. (nrfAH)", # nrfAH
    # C fixation
    ec ==  "1.2.7.4" ~ "16: CO dehydrogenase (cooS)", # WL pathway 
    ec =="2.3.1.169" ~ "17: Acetyl-CoA synthase (acs)", # WL pathway 
    ec == "4.1.1.39" ~ "18: RuBisCO", # Calvin cycle
    ec =="4.2.1.120" ~ "19: 4-hydroxybutyryl-CoA dehydratase (abfD)",
    ec == "6.4.1.3"  ~ "20: propionyl-CoA carboxylase (pccB)", # 3-Hydroxypropionate bi-cycle
    # H2 metabolism
    ec == "1.12.1.2" ~ "22: NAD-reducing hydrogenase (hoxfHY)",
    ec %in% c("1.12.99.6","1.12.2.1","1.12.7.2") ~ "23: NiFe hydrogenase (hydAB, hoxGK)"
  )) %>% 
  # Remove genes outside of above selection
  filter(!is.na(category)) %>%
  group_by(genome,gene) %>% summarise(n = n()) %>%
  inner_join(bintax, by = "genome") %>% filter(genome %in% names & n > 1) %>%
  select(taxon, gene, n, genome)
```




```{r marker genes - eggnog}
mg <- emapper %>%
  rename(ec = EC, gene = Preferred_name) %>%
  #select(genome, ec = EC, gene = Preferred_name) %>%
  mutate(category = case_when(
    # S metabolism
    ec == "2.7.7.4" ~ "03: Sulfate adenylyltransferase (sat)", # sat
    grepl("apr[A,B]", gene) ~ "04: Adenylylsulfate reductase (aprAB)", # aprAB
    grepl("asr[A-C]", gene) ~  "05: Anaerobic sulfite reductase (asrABC)", # asrABC
    grepl("sox[X,Y,Z,B]", gene) ~ "07: Thiosulfate ox. (soxABXYZ)", # soxA: soxBCD not present
    gene == "soeB" ~ "08: Sulfite dehydrogenase (soeB)", # soeB: soeAC not present
    gene == "sorB" ~ "08: Sulfite dehydrogenase (sorB)",
    # N metabolism
    ec == "1.18.6.1" ~ "09: Nitrogen fixation (nifKDH)", # nifKDH
    grepl("nrf[A,H]", gene) ~ "13: Dissimilatory nitrate red. (nrfAH)", # nrfAH
    # C fixation
    ec ==  "1.2.7.4" ~ "16: CO dehydrogenase (cooS)", # WL pathway 
    ec =="2.3.1.169" ~ "17: Acetyl-CoA synthase (acs)", # WL pathway 
    ec == "4.1.1.39" ~ "18: RuBisCO", # Calvin cycle
    ec =="4.2.1.120" ~ "19: 4-hydroxybutyryl-CoA dehydratase (abfD)",
    ec == "6.4.1.3"  ~ "20: propionyl-CoA carboxylase (pccB)", # 3-Hydroxypropionate bi-cycle
    # H2 metabolism
    ec == "1.12.1.2" ~ "22: NAD-reducing hydrogenase (hoxfHY)",
    ec %in% c("1.12.99.6","1.12.2.1","1.12.7.2") ~ "23: NiFe hydrogenase (hydAB, hoxGK)"
  )) %>% 
  # Remove genes outside of above selection
  filter(!is.na(category)) %>%
  inner_join(bintax, by = "genome") %>% filter(genome %in% names) %>%
  select(category, genome) %>% mutate(value = 1)
```


```{r marker genes - prokka}
# Key gene(s) for pathways
# M00376 3-Hydroxypropionate bi-cycle: EC:6.4.1.3 propionyl-CoA carboxylase
# M00375 Hydroxypropionate-hydroxybutylate cycle: EC:4.2.1.120 4-hydroxybutyryl-CoA dehydratase
# M00374 Dicarboxylate-hydroxybutyrate cycle: EC:4.2.1.120 4-hydroxybutyryl-CoA dehydratase

# Check for multiple gene copies
mg <- prokka %>%
  mutate(category = case_when(
    # S metabolism
    ec == "2.7.7.4" ~ "03: Sulfate adenylyltransferase (sat)", # sat
    grepl("apr[A,B]", gene) ~ "04: Adenylylsulfate reductase (aprAB)", # aprAB
    grepl("asr[A-C]", gene) ~ "05: Anaerobic sulfite reductase (asrABC)", # asrABC
    grepl("sox[X,Y,Z,B]", gene) ~ "07: Thiosulfate ox. (soxABXYZ)", # soxA: soxBCD not present
    gene == "soeB" ~ "08: Sulfite dehydrogenase (soeB)", # soeB: soeAC not present
    gene == "sorB" ~ "08: Sulfite dehydrogenase (sorB)",
    # N metabolism
    ec == "1.18.6.1" ~ "09: Nitrogen fixation (nifKDH)", # nifKDH
    grepl("nrf[A,H]", gene) ~ "13: Dissimilatory nitrate red. (nrfAH)", # nrfAH
    # C fixation
    ec ==  "1.2.7.4" ~ "16: CO dehydrogenase (cooS)", # WL pathway 
    ec =="2.3.1.169" ~ "17: Acetyl-CoA synthase (acs)", # WL pathway 
    ec == "4.1.1.39" ~ "18: RuBisCO", # Calvin cycle
    ec =="4.2.1.120" ~ "19: 4-hydroxybutyryl-CoA dehydratase (abfD)",
    ec == "6.4.1.3"  ~ "20: propionyl-CoA carboxylase (pccB)", # 3-Hydroxypropionate bi-cycle
    # H2 metabolism
    ec == "1.12.1.2" ~ "22: NAD-reducing hydrogenase (hoxfHY)",
    ec %in% c("1.12.99.6","1.12.2.1","1.12.7.2") ~ "23: NiFe hydrogenase (hydAB, hoxGK)"
  )) %>%
  # Remove genes outside of above selection
  filter(!is.na(category)) %>%
  # Selected only the bins from the names object
  inner_join(bintax, by = "genome") %>% filter(genome %in% names) %>%
  select(category, genome) %>% mutate(value = 1) %>%
  # Add Eggnog mapper query
  rbind(mg) %>%
  distinct() %>%
  # Assign a value (0/1) to each combination
  pivot_wider(names_from = "genome", values_from = "value", values_fill = 0) %>%
  pivot_longer(-1, names_to = "genome", values_to = "value") %>%
  # Add order
  mutate(order = substr(category, 1, 2)) %>%
  mutate(category = substr(category, 5, nchar(.))) %>% 
  arrange(order)
```


```{r plot abundance (p1) plus marker genes (p2)}
# Create labels voor de bins
taxlab <- bintax %>% 
  mutate(label = if_else(taxon != genus | is.na(genus), taxon, paste(taxon, "sp."))) %>%
  filter(genome %in% names) %>%
  mutate(label = ifelse(label == "Desulforudis sp.", "Ca. D. audaxviator", label)) %>%
  mutate(genome = factor(genome, names)) %>% arrange(genome) %>%
  pull(label)

p1 <- bintax %>%
  # Use subset of bins
  filter(genome %in% names) %>%
  # Genus in italics
  inner_join(coverm, by = "genome") %>%
  # Group tpm per taxon and standardize to 1
  group_by(genome, phylum) %>% summarise(tpm = sum(tpm) / 3e6, .groups = "drop") %>%
  mutate(genome = factor(genome, names)) %>%
  mutate(Abundance = "Abundance") %>%
  # Plot
  ggplot(aes(genome, Abundance, fill = tpm)) +
  geom_tile(colour = "white") +
  labs(x = "", y = "", fill = "") +
  scale_fill_gradient(
    name = expression(Coverage ~ (phantom()%*% 10^{6})),
    low = "#C5ECFD", high = "#034968") +
  scale_x_discrete(position = "top", labels = taxlab) + 
  theme_tidy(ratio = NULL) +
  coord_equal() +
  theme(
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 0),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )

p2 <- mg %>%
  mutate(genome = factor(genome, names)) %>%
  mutate(category = factor(category, rev(unique(mg$category)))) %>%
  mutate(value = factor(value, levels = c("1", "0"))) %>%
  # Plot
  ggplot(aes(genome, category, fill = value)) +
  geom_tile(colour = "white") +
  scale_fill_manual(
    values = c("1" = "#444431","0" = "#F0F1EB"), 
    labels = c("Present", "Absent")) +
  labs(x = "", y = "", fill = "") +
  scale_x_discrete(position = "top") + theme_tidy(ratio = NULL) + coord_equal() +
  theme(
    plot.margin = margin(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_blank(),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```


```{r kegg modules}
# Consider adding M00309: Entner-Doudoroff pathway in central C metabolism
# Below C fixation pathways all incomplete
# M00374 Dicarboxylate-hydroxybutyrate cycle
# M00375 Hydroxypropionate-hydroxybutylate cycle
# M00376 3-Hydroxypropionate bi-cycle

t <- kegg %>%
  mutate(reference = case_when(
    # S metabolism
    module == "M00596" ~ "Dissimilatory sulfate red.",
    module == "M00176" ~ "Assimilatory sulfate red.",
    # C fixation
    module == "M00167" ~ "Calvin cycle",
    module == "M00173" ~ "Reductive citrate cycle",
    module == "M00377" ~ "Wood-Ljungdahl pathway",    
    # C metabolism
    module == "M00001" ~ "Glycolysis",
    module == "M00008" ~ "Entner-Doudoroff pathway",
    module == "M00009" ~ "Citrate cycle"
  )) %>%
  filter(!is.na(reference)) %>% # Only keep the modules listed above
  # Add module to the reference for plotting
  mutate(reference = paste(module, ": ", reference, sep = "")) %>%
  # Format the completeness for plotting
  mutate(genes = gsub("/.*", "", completeness) %>% as.numeric()) %>%
  mutate(total = gsub(".*/", "", completeness) %>% as.numeric()) %>%
  mutate(score = case_when(
    genes == total ~ "Complete",
    (total - genes) == 1 ~ "One block missing",
    .default = "Incomplete"
  )) %>%
  select(genome, reference, score) %>%
  # Assign a value for every genome-module combination
  pivot_wider(names_from = "reference", values_from = "score") %>%
  pivot_longer(-1, values_to = "score", names_to = "reference")
```


```{r plot kegg modules}
i <- c("M00596: Dissimilatory sulfate red.",
       "M00176: Assimilatory sulfate red.",
       "M00173: Reductive citrate cycle",
       "M00167: Calvin cycle",
       "M00377: Wood-Ljungdahl pathway",
       "M00001: Glycolysis",
       "M00008: Entner-Doudoroff pathway",
       "M00009: Citrate cycle"
       ) %>% rev()

p3 <- t %>%
  replace_na(list(score = "Absent")) %>%
  # Add taxon names
  inner_join(bintax, by = "genome") %>% 
  # Use subset of bins
  filter(genome %in% names) %>%
  mutate(genome = factor(genome, names)) %>%
  mutate(score = factor(score, levels = c("Complete","One block missing","Incomplete","Absent"))) %>%
  # Plot
  ggplot(aes(genome, 
             fct_relevel(reference, i), 
             fill = score)) +
  geom_tile(colour = "white") +
  scale_fill_manual(values = c("Complete" = "#444431",
                               "One block missing" = "#707151",
                               "Incomplete" = "#B6B79B",
                               "Absent" = "#F0F1EB"
                               )) +
  labs(x = "", y = "", fill = "") +
  scale_x_discrete(position = "top") + theme_tidy(ratio = NULL) + coord_equal() +
  theme(
    plot.margin = margin(),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_blank(),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```


```{r kegg pathways}
k <- emapper %>%
  mutate(reference = case_when(
    # Bacterial chemotaxis CheABCDRVWXYZBR
    grepl("K03407", KEGG_ko) ~ "Chemotaxis",
    grepl("K03408", KEGG_ko) ~ "Chemotaxis",
    grepl("K03409", KEGG_ko) ~ "Chemotaxis",
    grepl("K03410", KEGG_ko) ~ "Chemotaxis",
    grepl("K03411", KEGG_ko) ~ "Chemotaxis",
    grepl("K03412", KEGG_ko) ~ "Chemotaxis",
    grepl("K03413", KEGG_ko) ~ "Chemotaxis",
    grepl("K03414", KEGG_ko) ~ "Chemotaxis",
    grepl("K03415", KEGG_ko) ~ "Chemotaxis",
    grepl("K00575", KEGG_ko) ~ "Chemotaxis",
    grepl("K13924", KEGG_ko) ~ "Chemotaxis",
    grepl("K03406", KEGG_ko) ~ "Chemotaxis", # MCP
    grepl("K05874", KEGG_ko) ~ "Chemotaxis", # tsr
    grepl("K05875", KEGG_ko) ~ "Chemotaxis", # tar
    grepl("K05876", KEGG_ko) ~ "Chemotaxis", # trg
    grepl("K05877", KEGG_ko) ~ "Chemotaxis", # tap
    grepl("K03776", KEGG_ko) ~ "Chemotaxis", # Aer
    # Flagellar assembly
    grepl("fli", Preferred_name) ~ "Flagellar assembly",
    grepl("flg", Preferred_name) ~ "Flagellar assembly",
    grepl("mot", Preferred_name) ~ "Flagellar assembly",
    grepl("map02040", KEGG_Pathway) ~ "Flagellar assembly"
  )) %>%
  filter(!is.na(reference)) %>%
  select(genome, reference, ko = KEGG_ko) %>%
  group_by(genome, reference) %>% 
  summarise(n = unique(ko) %>% length(), .groups = "drop") %>%
  # Bin MEGAHIT-MetaBAT2Refined-VK-3516-cosc-f1_S23_L002.11 lacks any value
  rbind(list(genome = "MEGAHIT-MetaBAT2Refined-VK-3516-cosc-f1_S23_L002.11", reference = "Chemotaxis", n = 0)) %>%
  # Assign a value for every genome-module combination
  pivot_wider(names_from = "reference", values_from = "n") %>%
  pivot_longer(-1, values_to = "n", names_to = "reference") %>%
  # Normalize the score for every 
  #mutate(score = n / max(.$n, na.rm = TRUE)) %>% select(-n) %>%
  replace_na(list(n = 0))
```


```{r plot kegg pathways}
# Aspect ratio is y / x 

p4 <- k %>%
  # Add taxon names
  inner_join(bintax, by = "genome") %>% filter(genome %in% names) %>%
  ggplot(aes(x = fct_relevel(genome, names), 
             y = reference, 
             fill = n
         )) +
  geom_tile(colour = "white") +
  labs(x = "", y = "", fill = "") +
  scale_fill_gradient(
    name = "No. of genes (KOs)",
    low = "#C5ECFD", high = "#034968") +
  scale_x_discrete(position = "top") + theme_tidy(ratio = NULL) + coord_equal() +
  theme(
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_blank(),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```


```{r, fig.width=16/2.53}
library(patchwork)
p1 / p2 / p3 / p4 + plot_layout(guides = "collect") & 
  theme(plot.margin = margin(),
        panel.background = element_blank(),
        legend.key.height = unit(4, "mm"))
```

```{r}
ggsave("../figures/cmetabolism.pdf", width = 16, height = 18, units = "cm")
```

# 18S - eukaryotes

```{r}
euk <- read_tsv("eukaryotes/ASV_table.tsv", col_types = cols(.default = col_character(),
                                                                         count = col_integer())
                ) %>% filter(sample %in% c("WC-3642-S73-COSC-454F-981R-f1",
                                           "WC-3642-S74-COSC-454F-981R-f3",
                                           "WC-3642-S75-COSC-454F-981R-f5"
                       )) %>% 
  group_by(sample) %>%
  mutate(relab = count / sum(count)) %>% ungroup()

protist <- read_tsv("eukaryotes/ASV_tax.tsv", col_types = cols(.default = col_character())) %>%
  filter(seqid %in% euk$seqid)
```


```{r}
euk %>%
  inner_join(protist, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(6, mean_relab) -> t

protist %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  #mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r}
p3 <- euk %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop') %>%
  # Plot
  ggplot(aes(x = sample, 
             y = relab * 100, 
             fill = fct_relevel(topphylum, names(cols.euk))
             )) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank()
        ) +
  labs(x = "", y = "") +
  scale_fill_manual(values = cols.euk, name = "") + guides(fill = guide_legend(reverse = TRUE)) +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("text", x =  2, y = -3.5, label = "Marker gene (18S)", size = 7/.pt) +
  annotate("text", x =  0.55, y = Inf, label = "bold(c)", 
           size = 8/.pt, hjust = 0, vjust = 1, parse = TRUE)
```


# Rarefaction of 16 & 18S 

Function rarecurve draws a rarefaction curve for each row of the input data. The rarefaction curves are evaluated using the interval of step sample sizes, always including 1 and total sample size. If sample is specified, a vertical line is drawn at sample with horizontal lines for the rarefied species richnesses.

```{r rarefaction, fig.width=12/2.53}
# Perform the subsampling
rare <- seqtab %>%
  # Remove negative control
  filter(sample != "P28309_1064") %>%
  rbind(euk) %>%
  select(-relab) %>%
  pivot_wider(names_from = "seqid", values_from = "count", values_fill = 0) %>%
  column_to_rownames("sample") %>%
  vegan::rarecurve(sample = 10000, step = 100, tidy = TRUE) %>%
  tibble() %>%
  rename_with(tolower)

lab <- list(site = unique(rare$site), label = c("16S #1", "16S #2", "16S #3",
                                                "18S #1", "18S #2", "18S #3")
            ) %>% data.frame() %>%
  inner_join(rare, by = "site") %>%
  group_by(site) %>% slice_max(species) %>% ungroup() %>%
  mutate(sample = if_else(label == "18S #1", sample - 10000, sample))

# Plot the curve
p3 <- rare %>%
  ggplot(aes(sample, species, group = site)) + 
  geom_line() +
  geom_label(data = lab, aes(sample, species, label = label), size = 7/.pt) +
  labs(x = expression('No. of reads (' %*% '1000)'), y = "No. of ASVs") +
  theme_tidy() + 
  scale_x_continuous(labels = c("0", "50", "100", "150"), breaks = c(0, 50000, 100000, 150000)) +
  scale_y_continuous(labels = c("0", "500", "1,000", "1,500"), breaks = c(0, 500, 1000, 1500))
```


# Heat map D. audaxviator

```{r}
ani <- read_tsv("salida_matrix.txt", col_names = F, col_select = 1:3) %>%
  rename(x = X1, y = X2, ani = X3) %>%
  mutate(ani = round(ani, digits = 1))

y.lab <- c("Sweden (this study)", 
           "Sweden (this study)",
           "Russia (Kadnikov et al. 2018)",
           "South Africa (Magnabosco et al. 2015)",
           "South Africa (Chivian et al. 2008)", 
           "South Africa (Lau et al. 2016)", 
           "South Africa (Lau et al. 2014)", 
           "South Africa (Lau et al. 2014)", 
           "South Africa (Lau et al. 2016)", 
           "South Africa (Lau et al. 2016)")

x.lab <- c("Sweden (this study)",
           "Sweden (this study)",
           "South Africa (Lau et al. 2014)",
           "South Africa (Lau et al. 2016)", 
           "South Africa (Lau et al. 2014)",
           "South Africa (Lau et al. 2016)", 
           "South Africa (Lau et al. 2016)", 
           "South Africa (Magnabosco et al. 2015)",
           "Russia (Kadnikov et al. 2018)",
           "South Africa (Chivian et al. 2008)"
           )

ani %>% 
  pivot_wider(names_from = "y", values_from = "ani") %>%
  column_to_rownames("x") %>%
  pheatmap::pheatmap(
    color = colorRampPalette(c("#F0F1EB","#6A6A5A"))(100),
    cluster_cols = FALSE,
    border_color = "white",
    cellwidth = 20, cellheight = 20,
    treeheight_row = 20,
    drop_levels = TRUE,
    labels_row = x.lab, labels_col = y.lab,
    legend_breaks = c(98.5, 99, 99.5, 100),
    legend_labels = c("98.5 %", "99 %", "99.5 %", "100 %"),
    fontsize = 7,
    display_numbers = TRUE, fontsize_number = 7, number_format = "%g", number_color = "black",
    width = 14/2.53, height = 14/2.53,
    filename = "../figures/ani.pdf"
  )
```

