---
title: "Drilling COSC-2 microbiology"
subtitle: "Characterization of microbial communities"
author: "George"
date: "2023-02-15"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```


```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
```


### Files 16S ###

```{r read input files}
library(tidyverse)
# GTDB release 207 was used as reference database

seqtab <- read_tsv("ASV_table.tsv", col_types = cols(.default = col_character(), count = col_integer()))
gtdb <- read_tsv("ASV_tax.tsv", col_types = cols(.default = col_character()))
meta <- read_tsv("smd.txt", col_types = cols(.default = col_character()))

# Add relative abundances
seqtab <- seqtab %>%
  mutate(sample = gsub("_S[0-9].", "", sample)) %>%
  filter(sample %in% meta$sample) %>%
  group_by(sample) %>% 
  mutate(relab = count / sum(count)) %>% 
  ungroup()

gtdb <- gtdb %>% mutate(taxon = coalesce(genus,family,order,class,phylum,domain)) %>% select(-kingdom)
```

### Files metagenomes ###

```{r read meta-g}
# CoverM was used for mapping the trimmed reads to the genomes
# This file only contains the bins for this project (n = 27)
coverm <- read_tsv("coverm.tsv", col_types = cols(.default = col_character(), tpm = col_double()))

# Before running CheckM with the CPR market set, this objected contained 567 genomes
# After running CheckM with CPR markers 570 genomes
checkm <- read_tsv("bin-stats-checkm.tsv", show_col_types = F)
checkm <- checkm %>% filter(genome %in% coverm$genome)

# Taxonomy was assigned to bins with completeness > 50% and contamination > 5%
# Average completeness and contamination of Patescibacteria before CPR marker set is 65.8% (sd 7.09, n = 12) and 1.17% (sd 1.38)
# Using CPR market set these numbers are 91.8% (sd 7.95, n = 15) and 0.930 (sd 1.71)


bintax <- read_tsv("gtdbtk.summary.tsv", show_col_types = F) %>%
  select(genome = user_genome, classification) %>%
  mutate(classification = gsub("[a-z]__", "", classification)) %>%
  separate(classification, 
           into = c("domain","phylum","class","order","family","genus","species"), 
           sep = ";", fill = "right") %>%
  mutate(across(everything(), na_if, "")) %>%
  mutate(across(-1, gsub, pattern = "_[A-Z]", replacement = "")) %>%
  filter(genome %in% coverm$genome) %>% distinct() %>%
  # Remove placeholder names
  mutate(across(2:8, \(x) ifelse(grepl("[0-9]", x), NA, x))) %>%
  mutate(taxon = coalesce(genus, family, order, class, phylum))
```


```{r}
emapper <- data.frame()

for (file in list.files("emapper/", full.names = TRUE)) {
  name <- substr(file, 10, nchar(file)) %>% gsub("emapper.emapper.annotations", "", .)
  i <- read_tsv(file, show_col_types = F, comment = "##")
  i <- i %>% rename(query = 1) %>% mutate(genome = name) %>% distinct() %>% select(genome, everything()) %>% mutate(KEGG_ko = gsub("ko:", "", KEGG_ko))
  emapper <- emapper %>% rbind(i)
}

prokka <- data.frame()
for (file in list.files("prokka/", full.names = TRUE)) {
  name <- substr(file, 9, nchar(file)) %>% gsub(".tsv", "", .)
  i <- read_tsv(file, show_col_types = F, comment = "##")
  i <- i %>% mutate(genome = name) %>% distinct() %>% 
    select(genome, everything()) %>% 
    select(-locus_tag) %>% 
    rename(ec = EC_number)
  prokka <- prokka %>% rbind(i)
}

kegg <- data.frame()
for (file in list.files("keggmapper/", full.names = TRUE)) {
  genome <- substr(file, 13, nchar(file)) %>% gsub(".txt", "", .)
  i <- read_tsv(file, col_names = FALSE, show_col_types = FALSE)
  i <- i %>% filter(grepl("M00", X1)) %>%
    mutate(genome = genome) %>%
    mutate(module = substr(X1, 1, 6)) %>%
    mutate(completeness = sub(".*\\s", "", X1)) %>%
    mutate(completeness = gsub(")", "", completeness)) %>%
    mutate(name = substr(X1, 7, nchar(X1))) %>% select(-X1)
  kegg <- kegg %>% rbind(i)
}
```


```{r colours}
# Ranked in order of abundance
cols.phylum <- c(
  "Firmicutes" = "#A2A475",
  "Desulfobacterota" = "#02401B",
  "Actinobacteriota" = "#D69C4E",
  "Proteobacteria" = "#972D15",
  "Other" = "#899DA4"
  )

# Ranked in order of abundance
cols.taxon <- c(
  "Desulforudis" = "#A2A475",
  "Desulfitibacter" = "#FAEFD1",
  "Smithellaceae" = "#02401B",
  "Coriobacteriia" = "#D69C4E",
  "Pelotomaculaceae" = "#00A08A",
  "Dethiosulfatibacteraceae" = "#F2AD00",
  "Erysipelotrichaceae" = "#046C9A",
  "Other" = "#899DA4"
  )
```


### Explore community structure base on marker-genes ###


```{r barplot cosc filters phylum}
i <- c("P28309_1061","P28309_1062","P28309_1063","P28309_1064")
# The cosc filter contains 43 ASVs and 2189 reads

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & order != "uncultured bacterium") %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other"))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Order") +
  geom_col() +
  scale_fill_brewer(palette = "Paired") +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  4.45, y = 101, yend = 101) +
  annotate('text', x =  2, y = 103, size = 7 / .pt, label = "COSC filters", hjust = 0.5) +
  annotate('text', x =  4, y = 103, size = 7 / .pt, label = "Extraction c.", hjust = 0.5) +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        )
```


```{r export barplot DNA order}
ggsave("../figures/barplot-order.pdf", height = 10, width = 16, units = "cm")
```


```{r remove ASVs extraction controls}
seqs <- seqtab %>% filter(sample == "P28309_1064") %>% pull(seqid)

seqtab <- seqtab %>% filter(!seqid %in% i)
```


```{r barplot original biofilm}
i <- c("P28309_1061","P28309_1062","P28309_1063")
# The cosc filter contains 43 ASVs and 2189 reads

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & order != "uncultured bacterium") %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other"))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Order") +
  geom_col() +
  scale_fill_brewer(palette = "Paired") +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate('text', x =  2, y = 103, size = 7 / .pt, label = "COSC filters", hjust = 0.5) +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        )
```

```{r export barplot DNA phylum}
ggsave("../figures/barplot-order-original.pdf", height = 10, width = 16, units = "cm")
ggsave("../figures/barplot-order-original.png", height = 10, width = 16, units = "cm")
```


### Bar plot ###

```{r}
i <- c("VK-3516-f1","VK-3516-f3","VK-3516-f5","P28309_1061","P28309_1062","P28309_1063")

p1 <- coverm %>%
  # Prepare the data to be merged with the amplicon data
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(phylum = ifelse(!is.na(phylum) & phylum %in% t, phylum, "Other")) %>%
  # Sum abundance within phylum
  group_by(sample, phylum) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(phylum = factor(phylum, levels = names(cols.phylum))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, 
             fill = fct_rev(phylum))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  labs(x = "", y = "Relative abundance (%)") +
  scale_fill_manual(values = rev(cols.phylum), name = "") +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  0.55, y = Inf, label = "a", 
           size = 8/.pt, hjust = 0, vjust = 1) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt) +
  guides(fill = guide_legend(nrow = 3, reverse = TRUE))
```


```{r}
p2 <- coverm %>%
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(taxon = case_when(
    genus %in% c("Desulforudis","Desulfitibacter") ~ genus,
    family %in% c("Smithellaceae","Pelotomaculaceae","Dethiosulfatibacteraceae",
                  "Erysipelotrichaceae") ~ family,
    class == "Coriobacteriia" ~ class,
    .default = "Other")
    ) %>%
  # Sum abundance within phylum
  group_by(sample, taxon) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(taxon = factor(taxon, levels = names(cols.taxon))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, fill = fct_rev(taxon))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank()
        ) +
  labs(x = "", y = "") +
  scale_fill_manual(values = cols.taxon, name = "") +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt) +
  annotate("text", x =  0.55, y = Inf, label = "b", 
           size = 8/.pt, hjust = 0, vjust = 1) +  
  guides(fill = guide_legend(nrow = 4, reverse = TRUE))
```


```{r}
library(patchwork)
p1 + p2 & theme(
  legend.key.size = unit(3.5, "mm"),
  legend.spacing = margin(),
  legend.margin = margin(-12,0,-12,0),
  legend.box.margin = margin(),
  plot.margin = margin(),
  aspect.ratio = 1,
  legend.position = "bottom",
  panel.border = element_blank(),
  axis.line = element_line(linewidth = 0.4, colour = "black"),
  axis.title.y = element_text(margin = margin(0,-2,0,0))
)
```


```{r}
ggsave("../figures/barplot.pdf", width = 14, height = 10, units = "cm")
```


### Visualise stats of genomes ###


```{r}
i <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")
# Desulforudis audaxviator GC 0.6085068, genome size 2616697 bp
library(ggtext)

p1 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, GC, genome.size) %>%
  distinct() %>%
  mutate(genome.size = genome.size/1e6) %>%
  ggplot(aes(x = genome.size, 
             y = GC, 
             color = fct_relevel(phylum,i)
             )) +
  geom_smooth(method = "lm", formula = y ~ x, colour = "black", fill = "#d9d9d9", se = TRUE, linetype = 0) +
  geom_point(size = 2.5, shape = 1, stroke = 1) + 
  annotate("text", x = Inf, y = -Inf, 
           label = "paste(R ^ 2, \" = 0.06\")", 
           size = 7/.pt, vjust = -0.5, hjust = 1, parse = TRUE) +
  annotate("richtext", x = 2.8, y = 0.609, 
           label = "D. audaxviator: 2.6 Mbp", 
           size = 7/.pt, hjust = 0, label.color = "#A2A475") +
  labs(x = "Estimated genome size (Mbp)", y = "GC content (%)", color = "") +
  scale_x_continuous(breaks = c(1,2,3,4,5,6)) +
  scale_color_manual(values = cols.phylum, guide = "none") +
  theme_tidy() +
  theme(
    text = element_text(colour = "black"),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, "black")
  )
```



```{r plot bin stats, fig.width = 14/2.54}
names <- c(
  "completeness" = "Completeness (%)",
  "predicted.genes" = "No. predicted genes"
)

p2 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, completeness) %>%
  distinct() %>%
  ggplot(aes(x = fct_relevel(phylum,i), 
             y = completeness, 
             group = phylum, 
             colour = phylum, 
             fill = fct_relevel(phylum,i)
             )) +
  geom_boxplot(alpha = 0.5, key_glyph = "rect") +
  labs(x = "", y = "Completeness (%)", fill = "") +
  scale_colour_manual(values = cols.phylum, guide = "none") +
  scale_fill_manual(values = cols.phylum) +
  theme_tidy(legend = "bottom", ratio = 1.2) +
  theme(
    axis.line = element_line(linewidth = 0.4, "black"),
    panel.border = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    legend.key.size = unit(4, "mm"),
    axis.title.y = element_text(margin = margin()),
    plot.margin = margin()
  )
```


```{r ordination on function metagenomes}
nmds <- coverm %>%
  inner_join(emapper, by = "genome") %>% 
  filter(KEGG_ko != "-") %>% # Filter out NA
  group_by(genome, KEGG_ko) %>% summarise(tpm = sum(tpm), .groups = "drop") %>%
  pivot_wider(names_from = "KEGG_ko", values_from = "tpm", values_fill = 0) %>%
  column_to_rownames("genome") %>%
  vegan::metaMDS()

nmds.scores <- vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample")
```


```{r}
library(ggtext)
i <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")

p3 <- nmds.scores %>%
  rename(genome = sample) %>% inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  # Plot
  ggplot(aes(x = NMDS1, 
             y = NMDS2, 
             colour = fct_relevel(phylum,i))) +
  geom_vline(xintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  geom_hline(yintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  # Points for samples, coloured by nature
  geom_point(size = 2.5, shape = 1, stroke = 1) +
  theme_tidy() +
  scale_color_manual(values = cols.phylum, name = "", guide = "none") +
  annotate('text', x = Inf, y = -Inf, size = 7/.pt,
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = 1.05, vjust = -0.5) +
  annotate("richtext", x = -2.9, y = 3.69, 
           label = "Methanobacteriota", 
           size = 7/.pt, hjust = 0, label.color = "#899DA4", 
           label.padding = unit(c(0.1,0.1,0.1,0.1), "lines")) +
  theme(
    axis.title.y = element_text(margin = margin()),
    panel.background = element_blank(),
    panel.border = element_rect(linewidth = 0.4, colour = "black")
    )
```


```{r, fig.width = 18/2.54}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") + plot_annotation(tag_levels = "a") & 
  theme(
    plot.tag.position = c(0.1, 1.05),
    plot.tag = element_text(size = 7, colour = "black"),
    legend.position = "bottom",
    plot.margin = margin(),
    plot.background = element_blank())
```


```{r export bin stats}
ggsave("../figures/binstats.pdf", height = 10, width = 18, units = "cm")
```


### Heatmap ###


```{r pathways}
i <- prokka %>%
  mutate(category = case_when(
    # S red
    ec == "2.7.7.4" ~ "03: Sulfate adenylyltransferase (sat)", # sat
    grepl("apr[A-B]", gene) ~ "04: Adenylylsulfate reductase (aprAB)", # aprAB
    COG %in% c("COG1145","COG0543","COG2221") ~  "05: Anaerobic sulfite reductase (asrABC)", # asrABC
    gene == "phsA" ~ "06: Thiosulfate reductase (phsA)",
    # S ox
    grepl("dsrE", gene) ~ "07: Dissimilatory sulfite reductase (dsrE)",
    grepl("sox[A,B]", gene) ~ "07: Thiosulfate oxidation (soxAB)", # soxAB: soxCD not present
    gene == "soeB" ~ "08: Sulfite dehydrogenase (soeB)", # soeB: soeAC not present
    gene == "sorB" ~ "08: Sulfite dehydrogenase (sorB)",
    # N metabolism
    grepl("nif[K,D,H]", gene) ~ "09: Nitrogen fixation (nifKDH)", # nifKDH
    grepl("nar[G,H,I]", gene) ~ "10: Nitrate reductase (narGHI)", # narGHI
    grepl("nap[A,B]", gene) ~ "11: Nitrate reductase (napAB)", # napAB
    gene == "nirD" ~ "12: DNRA (nirD)", # nirD: nirB not present
    grepl("nrf[A,H]", gene) ~ "13: DNRA (nrfAH)", # nrfAH
    grepl("nir[K,S]", gene) ~ "14: Nitrite reductase (nirKS)", # nirKS
    gene == "nosZ" ~ "15: Nitrous-oxide reductase (nosz)", # nosZ
    # C fixation
    ec == "1.2.7.4" ~ "16: Wood-Ljungdahl (cooS)", # cooS
    ec == "4.1.1.39" ~ "17: Reductive pentosa phosphate (rubisco)", # Rubisco
    ec == "2.7.1.19" ~ "18: Reductive pentosa phosphate (PRK)", # PRK
    # Organic C
    ec == "6.2.1.1" ~ "19: Acetyl-CoA synthetase (acsA)", # ACSS
    ec == "1.1.1.1" ~ "20: Alcohol dehydrogenase (adh)", # Alcohol dehydrogenase
  )) %>% filter(!is.na(category)) %>%
  inner_join(bintax, by = "genome") %>%
  select(category, taxon) %>% mutate(presence = 1) %>%
  distinct() %>%
  # Assign a value (0/1) to each combination
  pivot_wider(names_from = "taxon", values_from = "presence", values_fill = 0) %>%
  pivot_longer(-1, names_to = "taxon", values_to = "presence")
```


```{r}
names <- bintax %>%
  inner_join(coverm, by = "genome") %>%
  group_by(taxon, phylum) %>% summarise(tpm = sum(tpm)) %>%
  arrange(desc(tpm)) 

i <- i %>%
  mutate(order = gsub(": .*", "", category)) %>%
  mutate(category = substr(category, 5,nchar(category)))
  
```

```{r}
annotation <- names %>%
  column_to_rownames("taxon") %>%
  mutate("Abundance (tpm)" = tpm) %>% rename(Phylum = phylum) %>% select(-tpm)

cols <- list(
  Phylum = c(Firmicutes = "#A2A475", Desulfobacterota = "#02401B", Actinobacteriota = "#D69C4E",
             Proteobacteria = "#972D15", Bacteroidota = "#046C9A", Chloroflexota = "#00A08A",
             Methanobacteriota = "#FAEFD1"),
  "Abundance (tpm)" = c(low = "#C6DBEF", high = "#08519C")
)

df<- i %>%
  arrange(order) %>%
  select(-order) %>%
  pivot_wider(names_from = "taxon", values_from = "presence", values_fill = 0) %>%
  column_to_rownames("category")

pheatmap::pheatmap(df[, names$taxon],
    border_color = "white",
    cluster_rows = F, cluster_cols = F,
    legend_breaks = c(0, 1), color = c("#C6DBEF","#08519C"), legend_labels = c("Absent","Present"),
    fontsize = 7,
    annotation_col = annotation,
    annotation_colors = cols
  )
```




```{r}
i %>%
  ggplot(aes(x = fct_relevel(taxon, names), 
             y = fct_reorder(category, order, .desc = TRUE), 
             fill = presence)) +
  geom_tile(colour = "white") +
  scale_fill_gradient(low = "white", high = "steelblue") +
  labs(x = "", y = "") +
  scale_x_discrete(position = "top") + theme_tidy() + coord_equal() +
  theme(
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 0),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```


```{r kegg modules}
t <- kegg %>%
  mutate(reference = case_when(
    module == "M00001" ~ "Glycolysis",
    module == "M00307" ~ "Pyruvate ox.",
    module == "M00165" ~ "Calvin cycle",
    module == "M00596" ~ "Dissimilatory sulfate red.",
    module == "M00173" ~ "Reductive citrate cycle",
    module == "M00377" ~ "Reductive acetyl-CoA pathway",
    module == "M00357" ~ "Methanogenesis",
    module == "M00175" ~ "Nitrogen fixation",
    module == "M00531" ~ "Assimilatory nitrate red.",
    module == "M00530" ~ "Dissimilatory nitrate red.",
    module == "M00529" ~ "Denitrification",
    module == "M00804" ~ "Complete nitrification",
    module == "M00176" ~ "Assimilatory sulfate red."
  )) %>%
  filter(!is.na(reference)) %>% # Only keep the modules listed above
  # Add module to the reference for plotting
  mutate(reference = paste(module, ": ", reference, sep = "")) %>%
  # Format the completeness for plotting
  mutate(genes = gsub("/.*", "", completeness) %>% as.numeric()) %>%
  mutate(total = gsub(".*/", "", completeness) %>% as.numeric()) %>%
  mutate(score = case_when(
    genes == total ~ "Complete",
    (total - genes) == 1 ~ "Near-complete",
    .default = "Incomplete"
  )) %>%
  select(genome, reference, score) %>%
  # Assign a value for every genome-module combination
  pivot_wider(names_from = "reference", values_from = "score") %>%
  pivot_longer(-1, values_to = "score", names_to = "reference")
```


```{r}
t %>%
  # Add taxon names
  inner_join(bintax, by = "genome") %>%
  ggplot(aes(x = fct_relevel(taxon, names$taxon), 
             y = reference, 
             fill = fct_relevel(score, c("Incomplete", "Near-complete", "Complete")))) +
  geom_tile(colour = "white") +
  scale_fill_manual(values = c("Incomplete" = "#D9DAC7",
                               "Near-complete" = "#A2A475",
                               "Complete" = "#616246"), na.value = "#D9DAC7") +
  labs(x = "", y = "", fill = "") +
  scale_x_discrete(position = "top") + theme_tidy() +
  theme(
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 0),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```


```{r}
k <- emapper %>%
  mutate(process = case_when(
    grepl("ko02030", KEGG_Pathway) ~ "Chemotaxis",
    grepl("ko02040", KEGG_Pathway) ~ "Flagellar assembly",
    grepl("ko02024", KEGG_Pathway) ~ "Quorum sensing"
  )) %>%
  filter(!is.na(process)) %>%
  select(genome, process, ko = KEGG_ko) %>%
  group_by(genome, process) %>% 
  summarise(n = unique(ko) %>% length(), .groups = "drop") %>%
  # Assign a value for every genome-module combination
  pivot_wider(names_from = "process", values_from = "n") %>%
  pivot_longer(-1, values_to = "n", names_to = "process")
```

```{r}
k %>%
  # Normalize each count to 1
  mutate(n = n / max(k$n, na.rm = TRUE)) %>%
  # Add taxon names
  inner_join(bintax, by = "genome") %>%
  ggplot(aes(x = fct_relevel(taxon, names$taxon), 
             y = process, 
             fill = n
         )) +
  geom_tile(colour = "white", height = 0.2) +
  labs(x = "", y = "", fill = "") +
  scale_x_discrete(position = "top") + theme_tidy() +
  theme(
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.x = element_text(angle = 90, hjust = 0),
    panel.grid.minor = element_line(colour = "white", linewidth = 2)
  )
```

