---
title: "Drilling COSC-2 microbiology"
subtitle: "Characterization of microbial communities"
author: "George"
date: "2023-02-15"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```


```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
```


### Files 16S ###

```{r read input files}
library(tidyverse)
# GTDB release 207 was used as reference database

seqtab <- read_tsv("ASV_table.tsv", col_types = cols(.default = col_character(), count = col_integer()))
gtdb <- read_tsv("ASV_tax.tsv", col_types = cols(.default = col_character()))
meta <- read_tsv("smd.txt", col_types = cols(.default = col_character()))

# Add relative abundances
seqtab <- seqtab %>%
  mutate(sample = gsub("_S[0-9].", "", sample)) %>%
  filter(sample %in% meta$sample) %>%
  group_by(sample) %>% 
  mutate(relab = count / sum(count)) %>% 
  ungroup()

gtdb <- gtdb %>% mutate(taxon = coalesce(genus,family,order,class,phylum,domain)) %>% select(-kingdom)
```

### Files metagenomes ###

```{r read meta-g}
# CoverM was used for mapping the trimmed reads to the genomes
# This file only contains the bins for this project (n = 27)
coverm <- read_tsv("coverm.tsv", col_types = cols(.default = col_character(), tpm = col_double()))

# Before running CheckM with the CPR market set, this objected contained 567 genomes
# After running CheckM with CPR markers 570 genomes
checkm <- read_tsv("bin-stats-checkm.tsv", show_col_types = F)
checkm <- checkm %>% filter(genome %in% coverm$genome)

# Taxonomy was assigned to bins with completeness > 50% and contamination > 5%
# Average completeness and contamination of Patescibacteria before CPR marker set is 65.8% (sd 7.09, n = 12) and 1.17% (sd 1.38)
# Using CPR market set these numbers are 91.8% (sd 7.95, n = 15) and 0.930 (sd 1.71)


bintax <- read_tsv("gtdbtk.summary.tsv", show_col_types = F) %>%
  select(genome = user_genome, classification) %>%
  mutate(classification = gsub("[a-z]__", "", classification)) %>%
  separate(classification, 
           into = c("domain","phylum","class","order","family","genus","species"), 
           sep = ";", fill = "right") %>%
  mutate(across(everything(), na_if, "")) %>%
  mutate(across(-1, gsub, pattern = "_[A-Z]", replacement = "")) %>%
  filter(genome %in% coverm$genome) %>% distinct() %>%
  # Remove placeholder names
  mutate(across(2:8, \(x) ifelse(grepl("[0-9]", x), NA, x))) %>%
  mutate(taxon = coalesce(genus, family, order, class, phylum))
```


```{r}
emapper <- data.frame()

for (file in list.files("emapper/", full.names = TRUE)) {
  name <- substr(file, 10, nchar(file)) %>% gsub("emapper.emapper.annotations", "", .)
  i <- read_tsv(file, show_col_types = F, comment = "##")
  i <- i %>% rename(query = 1) %>% mutate(genome = name) %>% distinct() %>% select(genome, everything())
  emapper <- emapper %>% rbind(i)
}
```


```{r colours}
# Ranked in order of abundance
cols.phylum <- c(
  "Firmicutes" = "#A2A475",
  "Desulfobacterota" = "#02401B",
  "Actinobacteriota" = "#D69C4E",
  "Proteobacteria" = "#972D15",
  "Other" = "#899DA4"
  )

# Ranked in order of abundance
cols <- c(
  "Patescibacteria" = "#D69C4E",
  "Desulfobacterota" = "#F2AD00",
  "Caldatribacteriota" = "#972D15",
  "Chloroflexota" = "#A2A475",
  "Acidobacteriota" = "#02401B",
  "Omnitrophota" = "#00A08A",
  "Nitrospirota" = "#046C9A",
  "Campylobacterota" = "#899DA4",
  "Other" = "#C7B19C",
  "Unidentified" = "#FAEFD1"
)
```


### Explore community structure base on marker-genes ###

```{r barplot cosc filters phylum}
i <- c("P28309_1061","P28309_1062","P28309_1063","P28309_1064")
# The cosc filter contains 43 ASVs and 2189 reads

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & order != "uncultured bacterium") %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other"))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Order") +
  geom_col() +
  scale_fill_brewer(palette = "Paired") +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate("segment", x =  3.55, xend =  4.45, y = 101, yend = 101) +
  annotate('text', x =  2, y = 103, size = 7 / .pt, label = "COSC filters", hjust = 0.5) +
  annotate('text', x =  4, y = 103, size = 7 / .pt, label = "Extraction c.", hjust = 0.5) +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        )
```
```{r export barplot DNA order}
ggsave("../figures/barplot-order.pdf", height = 10, width = 16, units = "cm")
```


```{r remove ASVs extraction controls}
seqs <- seqtab %>% filter(sample == "P28309_1064") %>% pull(seqid)

seqtab <- seqtab %>% filter(!seqid %in% i)
```


```{r barplot original biofilm}
i <- c("P28309_1061","P28309_1062","P28309_1063")
# The cosc filter contains 43 ASVs and 2189 reads

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(gtdb, by = "seqid") %>%
  group_by(order, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(order) & order != "uncultured bacterium") %>%
  top_n(10, mean_relab) -> t

gtdb %>%
  left_join(t %>% transmute(order, topphylum = order), by = "order") %>%
  mutate(topphylum = if_else(is.na(order), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, i),
             y = relab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other"))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Order") +
  geom_col() +
  scale_fill_brewer(palette = "Paired") +
  annotate("segment", x =  0.55, xend =  3.45, y = 101, yend = 101) +
  annotate('text', x =  2, y = 103, size = 7 / .pt, label = "COSC filters", hjust = 0.5) +
  theme_barplot(fontsize = 7) + 
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank(),
        axis.ticks.y = element_line(), plot.margin = margin(0,0,0,0),
        axis.text.y = element_text(face = "bold"),
        legend.position = "right", legend.title = element_text(face = "bold", size = 7),
        panel.border = element_rect(size = 1, fill = NA)
        )
```

```{r export barplot DNA phylum}
ggsave("../figures/barplot-order-original.pdf", height = 10, width = 16, units = "cm")
ggsave("../figures/barplot-order-original.png", height = 10, width = 16, units = "cm")
```


### Pie ###

```{r}
# Select most abundant taxa from metagenomes
i <- c("P28309_1061","P28309_1062","P28309_1063",
       "VK-3516-f1","VK-3516-f3","VK-3516-f5")

tax <- bintax %>% 
  rename(seqid = genome) %>%
  rbind(gtdb)

coverm %>%
  select(sample, seqid = genome, relab = tpm) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(phylum = ifelse(!is.na(phylum) & phylum %in% t, phylum, "Other"))


  mutate(phylum = ifelse(!is.na(phylum) & !phylum %in% t, phylum, "Other"))
  group_by(taxon) %>% 
  summarise(tpm = sum(tpm)) %>% 
  slice_max(tpm, n = 11) %>%
  left_join(bintax[,c("taxon","phylum")], by = "taxon") %>%
  group_by(taxon) %>% add_tally() %>% ungroup() %>% 
  select(-tpm) %>% distinct() %>%
  rbind(list(taxon = "Other", phylum = "Other", n = 14))

# Modify the coverm data for merging with marker genes
i <- coverm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(tpm = tpm/10000) %>%
  mutate(type = "metagenome") %>%
  select(sample, type, class, family, genus, tpm)

# Merge the metagenomes with marker genes
i <- seqtab %>%
  mutate(relab = relab*100) %>%
  filter(sample %in% c("P28309_1061","P28309_1062","P28309_1063")) %>%
  inner_join(gtdb, by = "seqid") %>%
  mutate(type = "marker") %>%
  select(sample, type, class, family, genus, tpm = relab) %>%
  rbind(i) %>%
  mutate(taxon = case_when(
    genus == "Desulforudis" ~ "Desulforudis",
    genus == "Desulfitibacter" ~ "Desulfitibacter",
    family == "Smithellaceae" ~ "Smithellaceae",
    class == "Coriobacteriia" ~ "Coriobacteriia",
    family == "Pelotomaculaceae" ~ "Pelotomaculaceae",
    family == "Dethiosulfatibacteraceae" ~ "Dethiosulfatibacteraceae",
    family == "Erysipelotrichaceae" ~ "Erysipelotrichaceae",
    family == "Syntrophomonadaceae" ~ "Syntrophomonadaceae",
    genus == "Hydrogenophaga" ~ "Hydrogenophaga",
    genus == "Brevundimonas" ~ "Brevundimonas",
    genus == "Arachnia" ~ "Arachnia",
    #class == "Dethiobacteria" ~ "Dethiobacteria",
    #genus == "Desulfomicrobium" ~ "Desulfomicrobium",
    #genus == "Bellilinea" ~ "Bellilinea",
    .default = "Other"
  )) %>% group_by(sample, type, taxon) %>% summarise(tpm = sum(tpm), .groups = "drop")
```

```{r, fig.width = 12/2.54}
cols.taxon <- c(
  "Firmicutes" = "#A2A475",
  "Desulfobacterota" = "#02401B",
  "Actinobacteriota" = "#D69C4E",
  "Proteobacteria" = "#972D15",
  #"Chloroflexota" = "#046C9A",
  "Other" = "#899DA4"
  )

i %>%
  left_join(t, by = "taxon") %>%
  mutate(type = factor(type, levels = c("metagenome","marker"))) %>%
  ggplot(aes(x = tpm, 
             y = fct_relevel(taxon, rev(t$taxon)),
             color = fct_relevel(phylum, names(cols.taxon)), 
             fill = fct_relevel(phylum, names(cols.taxon))
             )
         ) +
  geom_boxplot(alpha = 0.5) + 
  scale_x_log10(limits = c(0.1,100), 
                labels = c(0.1, 1, 10, 100),
                breaks = c(0.1, 1, 10, 100)) +
  scale_y_discrete(labels = rev(paste0(t$taxon," (",t$n,")"))) +
  facet_wrap(~ type,
             scales = "free_x",
             labeller = as_labeller(c("metagenome" = "Metagenomes", "marker" = "Marker gene"))) +
  labs(x = "Relative abundance (%)", y = "") +
  scale_color_manual(values = cols.taxon) +
  scale_fill_manual(values = cols.taxon) +
  theme_tidy(ratio = 2, legend = "bottom") + guides(color = guide_legend(nrow = 2)) +
  theme(
    legend.title = element_blank(),
    legend.key.size = unit(3.5, "mm"),
    legend.box.margin = margin(),
    plot.margin = margin(),
    axis.ticks.y = element_blank(),
    axis.ticks.length = unit(0.2,"mm"),
    strip.background = element_blank(),
    panel.grid.major.x = element_line(linewidth = 0.2, colour = "grey", linetype = "dotted")
  )
```
```{r}
ggsave("../figures/boxplot.taxa.pdf", width = 12, height = 12, units = "cm")
```


```{r pie meta-g}
t <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")

p1 <- coverm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% t, phylum, "Other")) %>%
  group_by(phylum) %>% 
  summarise(tpm = sum(tpm), .groups = 'drop') %>%
  ggplot(aes(x = "", 
             y = tpm / 3, 
             fill = fct_relevel(phylum,rev(t))
             )
         ) +
  labs() + ggtitle("Metagenomes") +
  geom_bar(stat = "identity", width = 1, color = "white") +
  geom_label(x = 0.80, y = 440000, label = "88%", show.legend = F, size = 7/.pt, fill = "white", label.size = 0) +
  coord_polar("y", start = 0) +
  scale_fill_manual(values = cols.phylum, guide = "none") + theme_void()
```


```{r}
p2 <- seqtab %>%
  filter(sample %in% c("P28309_1061","P28309_1062","P28309_1063")) %>%
  inner_join(gtdb, by = "seqid") %>%
  mutate(phylum = ifelse(phylum %in% t, phylum, "Other")) %>%
  group_by(phylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop') %>%
  ggplot(aes(x = "", 
             y = relab / 3, 
             fill = fct_relevel(phylum,rev(t))
             )
         ) +
  labs(fill = "Phylum") + ggtitle("16S amplicons") +
  geom_bar(stat = "identity", width = 1, color = "white") +
  geom_label(x = 0.80, y = 0.445, label = "89%", show.legend = F, size = 7/.pt, fill = "white", label.size = 0) +
  coord_polar("y", start = 0) +
  theme_void(base_size = 7) + 
  scale_fill_manual(values = cols.phylum) + guides(fill = guide_legend(reverse = TRUE)) +
  theme(
    plot.title = element_text(size = 7, hjust = 0.5),
    plot.margin = margin(0,0,0,0),
    legend.key.size = unit(4, "mm"),
    legend.text = element_text(size = 7),
    legend.title = element_text(size = 7)) 
```


```{r top genera}
t <- coverm %>% 
  inner_join(bintax, by = "genome") %>% 
  group_by(taxon) %>% 
  summarise(tpm = sum(tpm)) %>% 
  slice_max(tpm, n = 7)
```

```{r}
cols.taxon <- c(
  "Desulforudis" = "#A2A475",
  "Desulfitibacter" = "#FAEFD1",
  "Smithellaceae" = "#02401B",
  "Coriobacteriia" = "#D69C4E",
  "Pelotomaculaceae" = "#00A08A",
  "Dethiosulfatibacteraceae" = "#F2AD00",
  "Erysipelotrichaceae" = "#046C9A",
  "Other" = "#899DA4"
  )

bintax %>%
  mutate(taxon = case_when(
    genus %in% c("Desulforudis","Desulfitibacter") ~ genus,
    family %in% c("Smithellaceae","Pelotomaculaceae","Dethiosulfatibacteraceae",
                  "Erysipelotrichaceae") ~ family,
    class == "Coriobacteriia" ~ class)
    ) %>%
  left_join(t, by = "taxon") %>% select(-tpm) %>%
  replace_na(list("taxon" = "Other")) -> taxref

p3 <- coverm %>%
  inner_join(taxref, by = "genome") %>%
  group_by(taxon) %>% 
  summarise(tpm = sum(tpm), .groups = 'drop') %>%
  ggplot(aes(x = "", 
             y = tpm / 3, 
             fill = fct_relevel(taxon, names(cols.taxon))
             )
         ) +
  geom_bar(stat = "identity", width = 1, color = "white") +
  coord_polar("y", start = 0) +
  geom_label(x = 0.80, y = 334594.2, label = "67%", show.legend = F, size = 7/.pt, fill = "white", label.size = 0) +
  scale_fill_manual(values = cols.taxon, guide = "none") +
  theme_void()
```


```{r}
gtdb %>%
  mutate(taxon = case_when(
    genus %in% c("Desulforudis","Desulfitibacter") ~ genus,
    family %in% c("Smithellaceae","Pelotomaculaceae","Dethiosulfatibacteraceae",
                  "Erysipelotrichaceae") ~ family,
    class == "Coriobacteriia" ~ class)
    ) %>%
  left_join(t, by = "taxon") %>% select(-tpm) %>%
  replace_na(list("taxon" = "Other")) -> taxref

p4 <- seqtab %>%
  filter(sample %in% c("P28309_1061","P28309_1062","P28309_1063")) %>%
  inner_join(taxref, by = "seqid") %>%
  group_by(taxon) %>% 
  summarise(relab = sum(relab), .groups = 'drop') %>%
  ggplot(aes(x = "", 
             y = relab / 3, 
             fill = fct_relevel(taxon, names(cols.taxon))
             )
         ) +
  labs(fill = "Highest resolution") +
  geom_bar(stat = "identity", width = 1, color = "white") +
  coord_polar("y", start = 0) +
  geom_label(x = 0.80, y = 0.2773844, label = "55%", show.legend = F, size = 7/.pt, fill = "white", label.size = 0) +
  scale_fill_manual(values = cols.taxon) + 
  guides(fill = guide_legend(reverse = TRUE)) +
  theme_void()
```


```{r, fig.width = 14/2.53}
library(patchwork)
(p1 + p2) / (p3 + p4) & theme_bw() &
  theme(
    plot.title = element_text(size = 7, colour = "black", hjust = 0.5),
    legend.key = element_blank(),
    legend.key.size = unit(4, "mm"),
    axis.text = element_blank(),
    axis.title = element_blank(),
    axis.line = element_blank(),
    axis.ticks = element_blank(),
    legend.title = element_text(size = 7),
    legend.text = element_text(size = 7),
    plot.background = element_blank(),
    panel.grid = element_blank(),
    panel.background = element_blank(),
    panel.border = element_blank(),
    plot.margin = margin()
  )
```

```{r export pie}
ggsave("../figures/pie.pdf", height = 10, width = 12, units = "cm")
```


```{r}
i <- c("VK-3516-f1","VK-3516-f3","VK-3516-f5","P28309_1061","P28309_1062","P28309_1063")

p1 <- coverm %>%
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(phylum = ifelse(!is.na(phylum) & phylum %in% t, phylum, "Other")) %>%
  # Sum abundance within phylum
  group_by(sample, phylum) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(phylum = factor(phylum, levels = names(cols.phylum))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, 
             fill = fct_rev(phylum))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  labs(x = "", y = "Relative abundance (%)") +
  scale_fill_manual(values = rev(cols.phylum), name = "") +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  0.55, y = Inf, label = "a", 
           size = 8/.pt, hjust = 0, vjust = 1) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt) +
  guides(fill = guide_legend(nrow = 3))
```

```{r}
p2 <- coverm %>%
  select(sample, seqid = genome, relab = tpm) %>%
  mutate(relab = relab / 1e6) %>%
  rbind(seqtab[,c("sample","seqid","relab")]) %>%
  filter(sample %in% i) %>%
  left_join(
    bintax %>% rename(seqid = genome) %>% rbind(gtdb), by = "seqid"
    ) %>%
  mutate(taxon = case_when(
    genus %in% c("Desulforudis","Desulfitibacter") ~ genus,
    family %in% c("Smithellaceae","Pelotomaculaceae","Dethiosulfatibacteraceae",
                  "Erysipelotrichaceae") ~ family,
    class == "Coriobacteriia" ~ class,
    .default = "Other")
    ) %>%
  # Sum abundance within phylum
  group_by(sample, taxon) %>% summarise(relab = sum(relab), .groups = "drop") %>%
  mutate(taxon = factor(taxon, levels = names(cols.taxon))) %>%
  # Plot
  ggplot(aes(x = fct_relevel(sample, i), 
             y = relab * 100, fill = fct_rev(taxon))) +
  geom_col() + 
  theme_tidy() + 
  theme(axis.text.x = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.x = element_blank(),
        axis.ticks.y = element_blank()
        ) +
  labs(x = "", y = "") +
  scale_fill_manual(values = cols.taxon, name = "") +
  annotate("segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate("segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate("text", x =  2, y = -3.5, label = "Metagenomes", size = 7/.pt) +
  annotate("text", x =  5, y = -3.5, label = "Marker gene (16S)", size = 7/.pt) +
  annotate("text", x =  0.55, y = Inf, label = "b", 
           size = 8/.pt, hjust = 0, vjust = 1) +  
  guides(fill = guide_legend(nrow = 4))
```

```{r}
library(patchwork)
p1 + p2 & theme(
  legend.key.size = unit(3.5, "mm"),
  legend.spacing = margin(),
  legend.margin = margin(-12,0,-12,0),
  legend.box.margin = margin(),
  plot.margin = margin(),
  aspect.ratio = 1,
  legend.position = "bottom",
  panel.border = element_blank(),
  axis.line = element_line(linewidth = 0.4, colour = "black"),
  axis.title.y = element_text(margin = margin(0,-2,0,0))
)
```
```{r}
ggsave("../figures/barplot.pdf", width = 14, height = 10, units = "cm")
```


### Visualise stats of genomes ###

```{r}
i <- c("Firmicutes","Desulfobacterota","Actinobacteriota","Proteobacteria","Other")
# Desulforudis audaxviator GC 0.6085068, genome size 2616697 bp
library(ggtext)

p1 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, GC, genome.size) %>%
  distinct() %>%
  mutate(genome.size = genome.size/1e6) %>%
  ggplot(aes(x = genome.size, 
             y = GC, 
             color = fct_relevel(phylum,i)
             )) +
  geom_smooth(method = "lm", formula = y ~ x, colour = "black", fill = "#d9d9d9", se = TRUE, linetype = 0) +
  geom_point(size = 2.5, shape = 1, stroke = 1) + 
  annotate("text", x = Inf, y = -Inf, 
           label = "paste(R ^ 2, \" = 0.06\")", 
           size = 7/.pt, vjust = -0.5, hjust = 1, parse = TRUE) +
  annotate("richtext", x = 2.8, y = 0.609, 
           label = "D. audaxviator: 2.6 Mbp", 
           size = 7/.pt, hjust = 0, label.color = "#A2A475") +
  labs(x = "Estimated genome size (Mbp)", y = "GC content (%)", color = "") +
  scale_x_continuous(breaks = c(1,2,3,4,5,6)) +
  scale_color_manual(values = cols.phylum, guide = "none") +
  theme_tidy() +
  theme(
    text = element_text(colour = "black"),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, "black")
  )
```



```{r plot bin stats, fig.width = 14/2.54}
names <- c(
  "completeness" = "Completeness (%)",
  "predicted.genes" = "No. predicted genes"
)

p2 <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  select(genome, phylum, completeness) %>%
  distinct() %>%
  ggplot(aes(x = fct_relevel(phylum,i), 
             y = completeness, 
             group = phylum, 
             colour = phylum, 
             fill = fct_relevel(phylum,i)
             )) +
  geom_boxplot(alpha = 0.5, key_glyph = "rect") +
  labs(x = "", y = "Completeness (%)", fill = "") +
  scale_colour_manual(values = cols.phylum, guide = "none") +
  scale_fill_manual(values = cols.phylum) +
  theme_tidy(legend = "bottom", ratio = 1.2) +
  theme(
    axis.line = element_line(linewidth = 0.4, "black"),
    panel.border = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    legend.key.size = unit(4, "mm"),
    axis.title.y = element_text(margin = margin()),
    plot.margin = margin()
  )
```


```{r ordination on function metagenomes}
nmds <- coverm %>%
  inner_join(emapper, by = "genome") %>% 
  filter(KEGG_ko != "-") %>% # Filter out NA
  group_by(genome, KEGG_ko) %>% summarise(tpm = sum(tpm), .groups = "drop") %>%
  pivot_wider(names_from = "KEGG_ko", values_from = "tpm", values_fill = 0) %>%
  column_to_rownames("genome") %>%
  vegan::metaMDS()

nmds.scores <- vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample")
```
```{r}
library(ggtext)

p3 <- nmds.scores %>%
  rename(genome = sample) %>% inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% i, phylum, "Other")) %>%
  # Plot
  ggplot(aes(x = NMDS1, 
             y = NMDS2, 
             colour = fct_relevel(phylum,i))) +
  geom_vline(xintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  geom_hline(yintercept = 0, linetype = 'dotted', linewidth = 0.5) +
  # Points for samples, coloured by nature
  geom_point(size = 2.5, shape = 1, stroke = 1) +
  theme_tidy() +
  scale_color_manual(values = cols.phylum, name = "", guide = "none") +
  annotate('text', x = Inf, y = -Inf, size = 7/.pt,
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = 1.05, vjust = -0.5) +
  annotate("richtext", x = -2.9, y = 3.69, 
           label = "Methanobacteriota", 
           size = 7/.pt, hjust = 0, label.color = "#899DA4", 
           label.padding = unit(c(0.1,0.1,0.1,0.1), "lines")) +
  theme(
    axis.title.y = element_text(margin = margin()),
    panel.background = element_blank(),
    panel.border = element_rect(linewidth = 0.4, colour = "black")
    )
```



```{r, fig.width = 18/2.54}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") + plot_annotation(tag_levels = "a") & 
  theme(
    plot.tag.position = c(0.1, 1.05),
    plot.tag = element_text(size = 7, colour = "black"),
    legend.position = "bottom",
    plot.margin = margin(),
    plot.background = element_blank())
```


```{r export bin stats}
ggsave("../figures/binstats.pdf", height = 10, width = 18, units = "cm")
```



